README.md

```diff
@@ -1,46 +1,214 @@
 ```
 
-Go to `golemrpc/examples/` directory and run one of the examples:
+# User guide
 
-```sh
-source venv/bin/activate
-cd golemrpc/examples
-python raspa_task.py
+## Basic example
+
+First set up a logging module::
+
+```python
+import logging
+logging.basicConfig(level=logging.INFO)
+```
+
+Then create and start an RPCComponent:
+
+```python
+from golemrpc.rpccomponent import RPCComponent
+rpc = RPCComponent(
+    host='35.158.100.160',
+    port=61000,
+    cli_secret_filepath='golemcli_aws.tck',
+    rpc_cert_filepath='rpc_cert_aws.pem'
+)
+rpc.start()
+```
+
+It will spawn a separate thread that will handle communication with Golem on `35.158.100.160:61000`. Now define a `dummy_task` function 
+to compute on Golem provider's machine:
+
+```python
+def dummy_task(args):
+    return 1 + args['b']
+```
+
+Create a golemrpc `CreateTask` message with task type `GLambda` (an abbreviation for Golem Lambda, a type of task that takes serialized Python function and dictionary as input and sends them to the provider for computation. For more information about messages and events see [this](https://github.com/golemfactory/golemrpc/blob/threaded/docs/messages.md)):
+
+```python
+message = {
+    'type': 'CreateTask',
+    'task': {
+        'type': 'GLambda',
+        'options': {
+            'method': dummy_task,
+            'args': {'b': 2},
+        },
+    }
+}
+```
+
+Send it to RPCComponent:
+
+```python
+rpc.post(message)
+```
+
+Now poll the RPC message queue for new events. First expected event is `TaskCreatedEvent`:
+
+```python
+task_created_event = rpc.poll(timeout=None)
+```
+
+It contains `task_id`assigned to this particular task by the requesting Golem node. One can use it to keep track of tasks in case there are more than one being computed at a time. Second message coming from RPCComponent should be `TaskResults` containing filepaths to actual results:
+
+```python
+task_results = rpc.poll(timeout=None)
+```
+
+By default there are three files listed in `TaskResults` message: `result.json`, `stderr.log` and `stdout.log`. Result returned by `dummy_task` function is serialized to a `result.json` of form:
+
+```json
+{
+    "data": 3,
+}
+```
+
+If there are any errors in user supplied function the JSON object will contain `error` key:
+
+```json
+{
+    "error": "Some error message"
+}
+```
+
+Pick result file, load it and verify the results:
+
+```python
+import json
+
+result_json_file = None
+
+for f in task_results['results']:
+    if f.endswith('result.json'):
+        result_json_file = f
+        break
+
+with open(result_json_file, 'r') as f:
+    result_json = json.load(f)
+assert result_json['data'] == (1 + 2)
+```
+
+## Big input files
+
+By default `CreateTask` message cannot exceed 0.5MB. One must use `resources` task's field instead function `args` to supply bigger file inputs. Files listed in `resources` will be laid out in `/golem/resources` directory which is accessible from user supplied function, e.g.:
```

Comment:
```suggestion
By default, `CreateTask` message cannot exceed 0.5MB. One must use `resources` task's field instead of function `args` in order to supply bigger file inputs. Files listed in `resources` will be saved in `/golem/resources` directory (with the path structure preserved) which is accessible from the user supplied function, e.g.:
```